metadata-ingestion/src/datahub/ingestion/source/gc/soft_deleted_entity_cleanup.py

```diff
@@ -103,48 +127,53 @@ def __init__(
         self.config = config
         self.report = report
         self.dry_run = dry_run
+        self.start_time = 0.0
+        self._report_lock: Lock = Lock()
+        self.last_print_time = 0.0
+
+    def _increment_retained_count(self) -> None:
+        """Thread-safe method to update report fields"""
+        with self._report_lock:
+            self.report.num_soft_deleted_retained_due_to_age += 1
```

Comment:
This could be wrapped this into our own implementation of _atomic counter_, or (quick search in internet) use this one here https://github.com/doodspav/atomics